cli/azd/pkg/commands/pipeline/github_provider.go

```diff
@@ -742,7 +703,19 @@ func ensureGitHubLogin(ctx context.Context, ghCli github.GitHubCli, hostname str
 			return false, errors.New("interactive GitHub login declined; use `gh auth login` to log into GitHub")
 		}
 
+		ghGitProtocol, err := ghCli.GetGitProtocolType(ctx)
```

Comment:
Could this value change after `gh auth login`, where the user can set the procotol type?

Reply:
Yes, but if users change this later it does not affect.

What we are doing here is to ensure `git push` will work when the protocol is HTTPS. why?  Because git remote will be added in the form of `https://repo...` , so when we do `git push`, git will need a credential (token) as password or fail.

If the gh cli protocol has previously set to `ssh`, that means running `git push` will use thatever key is in `~/.ssh`